### PR TITLE
feat(dataform): support workspace_compilation_overrides in google_dataform_repository

### PR DESCRIPTION
**Pull request by shotarok**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/14737

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dataform: added field `workspace_compilation_overrides` to resource `google_dataform_repository` (beta)
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_dataform_repository_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  git_remote_settings {
+      url = google_sourcerepo_repository.git_repository.url
+      default_branch = "main"
+      authentication_token_secret_version = google_secret_manager_secret_version.secret_version.id
+  }
+
+  workspace_compilation_overrides {
+    default_database = "database_v2"
+    schema_suffix = "_suffix_v2"
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_dataform_repository_test.go:122]:
```suggestion
```

Can we test removing one of these values?

Comment (shotarok) [reply mmv1/third_party/terraform/tests/resource_dataform_repository_test.go:121]: I deleted the field: `default_database` in [ca92493](https://github.com/GoogleCloudPlatform/magic-modules/pull/8074/commits/ca92493b72c54ed4145060258b1acf8fa20695bb).

Comment (shotarok) [reply mmv1/third_party/terraform/tests/resource_dataform_repository_test.go:121]: In addition to that, I updated the imported module to fix the error on terraform-provider-google-build-and-unit-tests with 5417e7f.

Comment (shotarok) [reply mmv1/third_party/terraform/tests/resource_dataform_repository_test.go:121]: @rileykarson google_dataform_respository is a beta resource. Therefore, testAccCheckDataformRepositoryDestroyProducer was not generated along with an auto-generated test in ` terraform-provider-google-build-and-unit-tests`

```
Error: google/resource_dataform_repository_test.go:23:2[9](https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/5158311281/jobs/9291839419#step:8:10): undefined: testAccCheckDataformRepositoryDestroyProducer
```
from https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/5158311281/jobs/9291839419

The unit tests passed locally under terraform-provider-google-beta. I'll copy the function with a different name as mitigation.

Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_dataform_repository_test.go:121]: Oh, we'll need to guard the test file: https://googlecloudplatform.github.io/magic-modules/how-to/add-handwritten-test/#testing-beta-features

Comment (shotarok) [reply mmv1/third_party/terraform/tests/resource_dataform_repository_test.go]: @rileykarson Thank you for your feedback! I added the guard in [aaa9bc1](https://github.com/GoogleCloudPlatform/magic-modules/pull/8074/commits/aaa9bc1867e537300d72a81de442602e266a37a5). I added `autogen_exception` too along with the other handwritten tests. Please let me know if you have a concern. Thanks

**Comment by shotarok** [comment]
@rileykarson Hi, when you have time could you please approve the cloud builder job? Thanks!

**Review by rileykarson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**